src/build/__init__.py

```diff
@@ -219,7 +219,38 @@ def check_dependencies(self, distribution, config_settings=None):
         dependencies = self.get_dependencies(distribution, config_settings).union(self.build_dependencies)
         return {u for d in dependencies for u in check_dependency(d)}
 
-    def build(self, distribution, outdir, config_settings=None):  # type: (str, str, Optional[ConfigSettings]) -> str
+    def prepare(self, distribution, outdir, config_settings=None):
+        # type: (str, str, Optional[ConfigSettings]) -> Optional[str]
+        """
+        Prepare metadata for a distribution.
+
+        :param distribution: Distribution to build (must be ``wheel``)
+        :param outdir: Output directory
+        :returns: The full path to the prepared metadata directory
+        """
+        prepare = getattr(self._hook, 'prepare_metadata_for_build_{}'.format(distribution))
+        outdir = os.path.abspath(outdir)
+
+        if os.path.exists(outdir):
+            if not os.path.isdir(outdir):
+                raise BuildException("Build path '{}' exists and is not a directory".format(outdir))
+        else:
+            os.mkdir(outdir)
+
+        try:
+            with _working_directory(self.srcdir):
+                basename = prepare(outdir, config_settings, _allow_fallback=False)  # type: str
+            path = os.path.join(outdir, basename)
+        except pep517.wrappers.BackendUnavailable:
+            raise BuildException("Backend '{}' is not available.".format(self._backend))
+        except pep517.wrappers.HookMissing:
+            return None
+        except Exception as e:  # noqa: E722
+            raise BuildBackendException('Backend operation failed: {!r}'.format(e))
+        return path
+
+    def build(self, distribution, outdir, metadata_directory=None, config_settings=None):
```

Comment:
`metadata_directory` -> `metadatadir` or `outdir` -> `output_directory` for consistency.

Reply:
PEP 517 names the arguments `wheel_directory` and `metadata_directory`. Would that work?

Reply:
Well, `build` works both with wheels and sdists, hence 'out' or 'output'.

Reply:
I’m going with `output_directory`.